lib/daemon.js

```diff
@@ -23,71 +23,68 @@ const commandsToFunctionNames = {
+    // once in a while.
+    setInterval(() => {
+      loglevel.debug('Making sure that the parent process ' +
+                    `(PID=${parentPid}) is still running.`);
+      try {
+        process.kill(parentPid, 0);
+      } catch (error) {
+        loglevel.info('Parent process seems to have died. Exiting.');
+        process.exit(1);
+      }
+    }, 30000);
+  }
 
-    WatchmanFileCache.getForWorkingDirectory(process.cwd()).initialize()
-      .then(() => {
-        loglevel.info(`WATCHMAN file cache is enabled for ${process.cwd()}`);
-      })
-      .catch((err: Object) => {
-        loglevel.info(
-          `WATCHMAN file cache is not available. Reason:\n${err.stack}`);
-      });
+  const rlInterface = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+    terminal: false,
+  });
 
-    if (parentPid) {
-      // Editor plugins should provide a `--parent-pid=<pid>` argument on startup,
-      // so that we can check that the daemon process hasn't turned into a zombie
-      // once in a while.
-      setInterval(() => {
-        loglevel.debug('Making sure that the parent process ' +
-                      `(PID=${parentPid}) is still running.`);
-        try {
-          process.kill(parentPid, 0);
-        } catch (error) {
-          loglevel.info('Parent process seems to have died. Exiting.');
-          process.exit(1);
-        }
-      }, 30000);
-    }
+  rlInterface.on('line', (jsonPayload: string) => {
+    loglevel.debug(`RECEIVED payload: ${jsonPayload}`);
 
-    const rlInterface = readline.createInterface({
-      input: process.stdin,
-      output: process.stdout,
-      terminal: false,
-    });
+    const payload = JSON.parse(jsonPayload);
 
-    rlInterface.on('line', (jsonPayload: string) => {
-      loglevel.debug(`RECEIVED payload: ${jsonPayload}`);
-      // The json payload is an array containing a job id and job details.
-      const payload = JSON.parse(jsonPayload);
-      const importer = new Importer(
-        payload.fileContent.split('\n'), payload.pathToFile);
+    const workingDirectory = findProjectRoot(payload.pathToFile);
+    loglevel.debug(
+      `Using ${workingDirectory} as project root for ${payload.pathToFile}`);
 
-      const functionName = commandsToFunctionNames[payload.command];
-      if (!functionName) {
-        const errorString =
-          `Unknown command: ${payload.command}. ` +
-          `Valid ones are ${Object.keys(commandsToFunctionNames).join(', ')}`;
-        loglevel.error(errorString);
-        const jsonResponse = JSON.stringify({ error: errorString });
-        process.stdout.write(`${jsonResponse}\n`);
-        return;
-      }
+    const importer = new Importer(
+      payload.fileContent.split('\n'),
+      payload.pathToFile,
+      workingDirectory
+    );
 
+    const functionName = commandsToFunctionNames[payload.command];
+    if (!functionName) {
+      const errorString =
+        `Unknown command: ${payload.command}. ` +
```

Comment:
That would be okay, but then you get a bunch of extra spaces in the middle. I was thinking more like this:

```js
const errorString = `Unknown command: ${payload.command}. Valid ones are ${Object.keys(commandsToFunctionNames).join(', ')}`;
```

But it doesn't really matter a whole lot